### ISSUE TITLE
Binaries not downloading?

### ISSUE THREAD

**Issue by RedDuckss** [vot/ffbinaries-node#9]
Found this module while looking for a way to convert mp4 to mp3, and it seemed useful. But I can't seem to get it to actually download anything?
```javascript
var ffbinariesPlatform = ffbinaries.detectPlatform();

createDirectory('./ffbinaries'); // Creates directory
 
ffbinaries.downloadFiles(ffbinariesPlatform, {destination:'./ffbinaries'}, () => {
  	console.log('Done');
});
```

This _seems_ to work, as it does log `Done` as expected. But nothing is saved to the `ffbinaries` folder. The folder exists, and no errors are thrown. Just doesn't seem to want to download the binaries there (or anywhere).

**Comment by vot** [comment]
Hi RedDuckss,

The syntax has changed slightly in version 1.0.0 and the platform is now a parameter in the options object (instead of being the first argument).

What I think you're after is something like this:
```
ffbinaries.downloadFiles('ffmpeg', {destination: './ffbinaries'}, (err, data) => {
  if (err) {
    console.log('Error', err);
  }
  console.log(data);
});
```

This will download only ffmpeg (you can specify array of components).
If you don't specify platform ffbinaries will automatically detect whatever platform is currently in use (equivalent of detecting it and passing it as an option).

Let me know if this works for you and the issue can be closed.

You can find a few examples here:
https://github.com/vot/ffbinaries-node/blob/master/examples/01-basic-usage.js

**Comment by vot** [comment]
Hi RedDuckss, I'm assuming you had no further problems with this? Is it ok to close the ticket?

**Comment by RedDuckss** [comment]
Hi! So sorry about that, I didn't see you had replied! My bad! Unfortunately the example you provided also does not work. If the destination folder does not exist, it is created, and the lib does not throw any errors and it logs the data as expected. However just like before, the destination folder is empty. Nothing seems to have been downloaded.

**Comment by vot** [comment]
Hi @RedDuckss,

No worries, I know it's easy to miss responses in issues. Also sorry for closing the issue without getting a confirmation from you, hopefully tagging you in this response will help with notifications :)

1) Which version of ffbinaries are you using in your project?
2) Do you have the same issue when downloading binaries through CLI? (to check that install globally with `npm i ffbinaries -g` and then run `ffbinaries ffmpeg` - could you please share output from that?)

**Comment by RedDuckss** [comment]
1. I have the latest version
2. Here's the output:

```
C:\Users\Admin\Documents\Programing\Electron\URTube>ffbinaries ffmpeg
ffbinaries downloader
------------------------------------
Platform not specified - downloading binaries for current platform.
Components: ffmpeg
Platform: windows-64
Destination: C:\Users\Admin\Documents\Programing\Electron\URTube
Files downloaded: ffmpeg-3.2-win-64.zip
------------------------------------
Binaries downloaded and extracted.
------------------------------------
```

It says the binaries have been downloaded an extracted, but when checking the destination file there's nothing relating to the binaries.

**Comment by vot** [comment]
Hi @RedDuckss, thanks for sharing the output.

I don't see download progress so I'm guessing you already have some cache stored in your system. Could you try wiping out the cache with `ffbinaries clearcache` and running that command again to see if invalid cache isn't a factor here?

**Comment by RedDuckss** [comment]
That did it!
```
C:\Users\Admin\Documents\Programing\Electron\URTube>ffbinaries clearcache
ffbinaries downloader
------------------------------------
Cache cleared

C:\Users\Admin\Documents\Programing\Electron\URTube>ffbinaries ffmpeg
ffbinaries downloader
------------------------------------
Platform not specified - downloading binaries for current platform.
Components: ffmpeg
Platform: windows-64
ffmpeg-3.2-win-64.zip: Received 1.311MB
ffmpeg-3.2-win-64.zip: Received 4.68MB
ffmpeg-3.2-win-64.zip: Received 8.947MB
ffmpeg-3.2-win-64.zip: Received 13.147MB
Destination: C:\Users\Admin\Documents\Programing\Electron\URTube
Files downloaded: ffmpeg-3.2-win-64.zip
------------------------------------
Binaries downloaded and extracted.
------------------------------------
```
And now `ffmpeg.exe` is in the destination folder. Is there any way to pragmatically clear the ffbinaries cache, so that I can make sure my app clears that cache before attempting to download the binaries, to prevent this from happening for other users? I am wanting to use this in an Electron app, which will be on different systems. So my goal here is to check if `ffmpeg` has already been downloaded, if not then clear the cache and download it for the users system.

**Comment by vot** [comment]
You can use `ffbinaries.clearCache()` programatically :)

In addition to that you can use `ffbinaries.getBinaryFilename(component, platform)` to supply filename for cross-platform `fs.exists` check to determine whether you need to download a file.

I think this will need to be treated as a bug in the long term, corrupted cache should not be stored in the first place. Many thanks for raising this :+1:

Vot

**Comment by vot** [comment]
Just got around to fixing this and published 1.0.3 to npm (see #11).
Now the temporary file will be stored with ".part" appended to its name and cache will be only persisted after download finishes.

This should resolve this for good - no need to call `clearCache()` from your code anymore :)

**Comment by RedDuckss** [comment]
Awesome! Thanks :)
